robots/commenter/main.go

```diff
@@ -221,9 +209,9 @@ func makeCommenter(comment string, useTemplate bool) func(meta) (string, error)
 	}
 }
 
-func run(c client, query, sort string, asc, random bool, commenter func(meta) (string, error), ceiling int) error {
+func run(c client, org, query, sort string, asc, random bool, commenter func(meta) (string, error), ceiling int) error {
 	log.Printf("Searching: %s", query)
-	issues, err := c.FindIssues(query, sort, asc)
+	issues, err := c.FindIssuesWithOrg(org, query, sort, asc)
```

Comment:
Interesting, I don't remember this. The need for the org arg could be avoided by extracting it from the query, it should always have one of `org:foo` or `repo:foo/bar`

Reply:
the JWT tokes signed for the GH app are signed only for the installation ID which is per org, the query doesn't really matter afaik, so you can only query a single org in a single job with GH apps

Reply:
The point is right now you have to specify the same org in both the query and the CLI arg, it would be a better UX if we extracted the org from the query so it doesn't have to be duplicated into the CLI arg.

Reply:
It has crossed my mind but:
a) you have to parse the query
b) the org can be implied be any of the `repo:` specs with the full name using `/`
c) with a) and b) together it would become complex and not intuitive (the error message can't just say use a single `org:` because of b) )

Reply:
I don't think its very complex, you just have to extract all `org:` and `repo:` blocks and for the `repo:` blocks then extract the org. If overall you get exactly one org, use that. If not, error out mentioning that you got too many or none. This could also be directly in the github client method to get rid of the mandatory org arg.

Reply:
I don't think this should be coupled to the GitHub search query syntax that can change over time. We'd always have to keep up. There is also set logic that you have to account for, like [is:issue team:kubernetes/sig-testing -org:kubernetes-sigs ](https://github.com/issues?q=is%3Aissue+team%3Akubernetes%2Fsig-testing+-org%3Akubernetes-sigs)